presto-main/src/main/java/com/facebook/presto/execution/MemoryRevokingScheduler.java

```diff
@@ -38,26 +42,54 @@
 
     private static final Ordering<SqlTask> ORDER_BY_CREATE_TIME = Ordering.natural().onResultOf(task -> task.getTaskInfo().getStats().getCreateTime());
     private final List<MemoryPool> memoryPools;
+    private final Supplier<? extends Collection<SqlTask>> currentTasksSupplier;
+    private final ScheduledExecutorService taskManagementExecutor;
     private final double memoryRevokingThreshold;
     private final double memoryRevokingTarget;
 
     @Inject
-    public MemoryRevokingScheduler(LocalMemoryManager localMemoryManager, FeaturesConfig config)
+    public MemoryRevokingScheduler(
+            LocalMemoryManager localMemoryManager,
+            SqlTaskManager sqlTaskManager,
```

Comment:
Injecting `SqlTaskManager` will not work as it is not bound explicitly.
Currently it is bound to interface `TaskManager` which does not have `getAllTasks` method.
You have to either:

 push the method up (which is probably ugly as exposes `SqlTask` which seems somewhat specific to `SqlTaskManager` via `TaskManager`.

or:

Explicitly bind `SqlTaskManager` and then bind `TaskManager` to that previous binding.

I would go for latter